### DISCUSSION TITLE
React Aria 3.7.0 -> 3.8.4 has increased bundle size by 50% ?

### DISCUSSION THREAD

**Discussion by fattslug** [adobe/react-spectrum#3100] [Q&A]
I can't seem to find a changelog anywhere.
Does anyone know why our bundle size has increased 50% after upgrading from 3.7.0 to 3.8.4?
We're using the @react-aria/interactions package specifically.

Thanks!

**Comment by snowystinger** [comment]
Can you tell us more about your setup/stack?


Also, odd question, but it does happen, so I'll ask, do you maybe have duplicates of any of our packages, maybe two different versions of one package?

**Reply by fattslug** [reply]
> Can you tell us more about your setup/stack?
> 
> Also, odd question, but it does happen, so I'll ask, do you maybe have duplicates of any of our packages, maybe two different versions of one package?

We're working in a monorepo using Lerna. Our builds are using a webpack config inherited from [design systems cli](https://github.com/intuit/design-systems-cli). The repo is a React component library.
No duplicates of the package. It's defined in one component's package.json of our monorepo, and I only see it defined once in the yarn.lock. 

Here's the yarn.lock definition (with some info *** removed):

```
"@react-aria/interactions@3.8.4":
  version "3.8.4"
  resolved "https://registry.npmjs.***.com:443/artifactory/api/npm/npm-***/@react-aria/interactions/-/interactions-3.8.4.tgz#f3704470b70f150b753bba01a8714f97d6299906"
  integrity sha1-***
  dependencies:
    "@babel/runtime" "^7.6.2"
    "@react-aria/utils" "^3.12.0"
    "@react-types/shared" "^3.12.0"

"@react-aria/ssr@^3.1.2":
  version "3.1.2"
  resolved "https://registry.npmjs.***.com:443/artifactory/api/npm/npm-***/@react-aria/ssr/-/ssr-3.1.2.tgz#665a6fd56385068c7417922af2d0d71b0618e52d"
  integrity sha1-***
  dependencies:
    "@babel/runtime" "^7.6.2"

"@react-aria/utils@^3.12.0":
  version "3.12.0"
  resolved "https://registry.npmjs.***.com:443/artifactory/api/npm/npm-***/@react-aria/utils/-/utils-3.12.0.tgz#39d0f37525e050356c4de725c85d9b10e7a5c0d9"
  integrity sha1-***
  dependencies:
    "@babel/runtime" "^7.6.2"
    "@react-aria/ssr" "^3.1.2"
    "@react-stately/utils" "^3.4.1"
    "@react-types/shared" "^3.12.0"
    clsx "^1.1.1"
```

Let me know if there's any other info you need that could be useful here.

**Reply by snowystinger** [reply]
I can't think of anything that would cause this without duplicates. Can you check if tree shaking is working appropriately? Can you tell what is causing the bloat?

You should be able to do this by building the previous version of your site, storing it in a temp directory. Then build the current version your site. Then run a diff tool comparing the two directories. Given how big a change it is, this should come up pretty quickly. You can also do this in a more visual way using something like webpack bundle analyzer plugin.

**Reply by fattslug** [reply]
My mistake - the bundle size increase wasn't due to this package. This can be closed.
Sorry for the fuss!
